include/aws/event-stream/event_stream_rpc.h

```diff
@@ -0,0 +1,59 @@
+#ifndef AWS_EVENT_STREAM_RPC_H
+#define AWS_EVENT_STREAM_RPC_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/event-stream/event_stream.h>
+
+/**
+ * :message-type header name
+ */
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_message_type_name;
+/**
+ * :message-flags header name
+ */
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_message_flags_name;
+/**
+ * :stream-id header name
+ */
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_stream_id_name;
+/**
+ * operation header name.
+ */
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_operation_name;
+
+enum aws_event_stream_rpc_message_type {
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_APPLICATION_MESSAGE,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_APPLICATION_ERROR,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PING,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PING_RESPONSE,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_CONNECT,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_CONNECT_ACK,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_INTERNAL_ERROR,
+
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_COUNT,
+};
+
+enum aws_event_stream_rpc_message_flag {
+    AWS_EVENT_STREAM_RPC_MESSAGE_FLAG_CONNECTION_ACCEPTED = 1,
+    AWS_EVENT_STREAM_RPC_MESSAGE_FLAG_TERMINATE_STREAM = 2,
+};
+
+struct aws_event_stream_rpc_message_args {
+    /** array of headers for an event-stream message. */
+    struct aws_event_stream_header_value_pair *headers;
+    /** number of headers in the headers array. */
+    size_t headers_count;
+    /** payload buffer for the message */
+    struct aws_byte_buf *payload;
```

Comment:
Document that the memory of the headers and payload is copied by send_message() call, so the user doesn't need to keep the `headers` and `payload` memory alive until the operation completes

right?